### PR TITLE
feat(translator): Add formatter support for OpenTelemetry access logger

### LINKED ISSUE #3570: OpenTelemetry access logger support extension formatters
follow https://github.com/envoyproxy/envoy/pull/34469

### PR DESCRIPTION
**Pull request by aoledk**
**What this PR does / why we need it**:

Add formatter support for OpenTelemetry access logger

**Which issue(s) this PR fixes**:

Fixes #3570

### CODE REVIEW

File: internal/xds/translator/accesslog.go [hunk 1]
```diff
@@ -278,6 +283,56 @@ func accessLogJSONFormatters(json map[string]string) []*cfgcore.TypedExtensionCo
...
+	if strings.Contains(body, celCommandOperator) {
+		cel = true
+	}
+
+	for _, value := range attributes {
+		if reqWithoutQuery && metadata && cel {
+			break
+		}
+
+		if strings.Contains(value, reqWithoutQueryCommandOperator) {
```
Comment (zirain) [inline internal/xds/translator/accesslog.go:306]:
```suggestion
		if !reqWithoutQuery && strings.Contains(value, reqWithoutQueryCommandOperator) {
```

File: internal/xds/translator/accesslog.go [hunk 2]
```diff
@@ -278,6 +283,56 @@ func accessLogJSONFormatters(json map[string]string) []*cfgcore.TypedExtensionCo
...
+			reqWithoutQuery = true
+		}
+
+		if strings.Contains(value, metadataCommandOperator) {
```
Comment (zirain) [inline internal/xds/translator/accesslog.go:310]: ditto

File: internal/xds/translator/accesslog.go [hunk 3]
```diff
@@ -278,6 +283,56 @@ func accessLogJSONFormatters(json map[string]string) []*cfgcore.TypedExtensionCo
...
+			metadata = true
+		}
+
+		if strings.Contains(value, celCommandOperator) {
```
Comment (zirain) [inline internal/xds/translator/accesslog.go:314]: ditto

**Comment by guydc** [comment]
cc @arkodg  - I think that this is another feature that will require Envoy 1.31 in EG 1.1.

**Comment by aoledk** [comment]
> cc @arkodg - I think that this is another feature that will require Envoy 1.31 in EG 1.1.

Yes, this feature in Envoy was merged into `main` branch last week by @zirain

**Review by guydc — APPROVED** [review]
LGTM, thanks!

**Review by zirain — APPROVED** [review]

**Comment by zirain** [comment]
it would be better to have an e2e test.

**Pull request merged**